src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -332,29 +332,19 @@ public static DetailNode getFirstChild(DetailNode node) {
      * @return true if node contains any node of type type among children on any deep level.
      */
     public static boolean containsInBranch(DetailNode node, int type) {
-        DetailNode curNode = node;
-        while (true) {
-
-            if (type == curNode.getType()) {
-                return true;
-            }
-
-            DetailNode toVisit = getFirstChild(curNode);
-            while (curNode != null && toVisit == null) {
-                toVisit = getNextSibling(curNode);
-                if (toVisit == null) {
-                    curNode = curNode.getParent();
+        boolean result = false;
+        if (type == node.getType()) {
+            result = true;
+        }
+        else {
+            for (DetailNode child : node.getChildren()) {
+                if (containsInBranch(child, type)) {
```

Comment:
Recursive calling will lead to stack overflow errors if tree happens to become deeply nested.
We had this problem in another area for one of our java grammar tests. I think we should avoid this.